reference_data/management/tests/update_gencode_tests.py

```diff
@@ -0,0 +1,220 @@
+        gene_info = _get_gene_infos(GeneInfo.objects.get(gene_id = 'ENSG00000284662'))
+        self.assertDictEqual(gene_info, {'start_grch37': 621059, 'chrom_grch37': u'1', 'coding_region_size_grch37': 936, 'gencode_release': 31, 'gencode_gene_type': u'protein_coding', 'gene_id': u'ENSG00000284662', 'gene_symbol': u'OR4F16', 'end_grch37': 622053, 'strand_grch37': u'-'})
+
+
+        self.assertEqual(TranscriptInfo.objects.all().count(), 2)
+        trans_info = _get_transcript_info(TranscriptInfo.objects.get(transcript_id = 'ENST00000456328'))
+        self.assertDictEqual(trans_info, {'start_grch37': 11869, 'end_grch37': 14409, 'strand_grch37': u'+', 'chrom_grch37': u'1', 'gene_id': u'ENSG00000223972'})
+        trans_info = _get_transcript_info(TranscriptInfo.objects.get(transcript_id = 'ENST00000332831'))
+        self.assertDictEqual(trans_info, {'start_grch37': 621059, 'end_grch37': 622053, 'strand_grch37': u'-', 'chrom_grch37': u'1', 'gene_id': u'ENSG00000284662'})
+
+        # Test normal command function with a --reset option
+        mock_logger.reset_mock()
+        call_command('update_gencode', '--reset', '--gencode-release=31', self.temp_file_path, '37')
+        calls = [
+            mock.call('Dropping the 2 existing TranscriptInfo entries'),
+            mock.call('Dropping the 50 existing GeneInfo entries'),
+            mock.call(
+                'Loading {} (genome version: 37)'.format(self.temp_file_path)),
+            mock.call('Creating 2 GeneInfo records'),
+            mock.call('Creating 2 TranscriptInfo records'),
+            mock.call('Done'),
+            mock.call('Stats: '),
+            mock.call('  genes_created: 2'),
+            mock.call('  transcripts_created: 2')
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        gene_infos = {gene.gene_id: {
+            'start_grch37': gene.start_grch37,
+            'chrom_grch37': gene.chrom_grch37,
+            'coding_region_size_grch37': gene.coding_region_size_grch37,
+            'gencode_release': gene.gencode_release,
+            'gencode_gene_type': gene.gencode_gene_type,
+            'gene_id': gene.gene_id,
+            'gene_symbol': gene.gene_symbol,
+            'end_grch37': gene.end_grch37,
+            'strand_grch37': gene.strand_grch37
+        } for gene in GeneInfo.objects.all()}
+        self.assertEqual(GeneInfo.objects.all().count(), 2)
+        gene_info = _get_gene_infos(GeneInfo.objects.get(gene_id = 'ENSG00000223972'))
+        self.assertDictEqual(gene_info, {'start_grch37': 11869, 'chrom_grch37': u'1', 'coding_region_size_grch37': 0, 'gencode_release': 31, 'gencode_gene_type': u'transcribed_unprocessed_pseudogene', 'gene_id': u'ENSG00000223972', 'gene_symbol': u'DDX11L1', 'end_grch37': 14409, 'strand_grch37': u'+'})
```

Comment:
Dictionaries are harder to read than line-by-line, and when something changes and they fail it is much harder to figure out what is different. If you already have a dictionary it makes sense to compare dictionaries, but it does not make sense to create 2 dictionaries just for the purpose of comparing them. Please do a line by line assertion as I asked. If you are really worried about copy-paste you could make a helper function that looks something like this, but its not neccessary:
```
def _assert_same_gene(gene, start_grch37=None, chrom_grch37=None, ...):
   self.assertEqual(gene. start_grch37, start_grch37)
   self.assertEqual(gene. chrom_grch37, chrom_grch37)
   ...
```

Reply:
Okay, I can do line by line tests for some of their fields.